lib/compat/wordpress-6.0/class-wp-webfonts.php

```diff
@@ -51,12 +69,174 @@ public function init() {
 			$this->stylesheet_handle = 'webfonts';
 			$hook                    = 'wp_enqueue_scripts';
 		}
+
+		add_action( 'init', array( $this, 'load_used_webfonts' ) );
+
+		add_action( 'switch_theme', array( $this, 'update_webfonts_used_by_templates' ) );
+
+		add_action( 'save_post_wp_template', array( $this, 'save_used_webfonts_for_template' ), 10, 2 );
+		add_action( 'save_post_wp_template_part', array( $this, 'update_webfonts_used_by_templates' ), 10, 2 );
+
 		add_action( $hook, array( $this, 'generate_and_enqueue_styles' ) );
 
 		// Enqueue webfonts in the block editor.
 		add_action( 'admin_init', array( $this, 'generate_and_enqueue_editor_styles' ) );
 	}
 
+	/**
+	 * Set list of used fonts in the current page.
+	 *
+	 * @return void
+	 */
+	public function load_used_webfonts() {
+		self::$used_webfonts = $this->get_globally_used_webfonts();
+		$used_webfonts       = get_option( self::$webfont_cache_option, array() );
+
+		foreach ( $used_webfonts as $template => $webfonts ) {
+			add_filter(
+				$template . '_template',
+				function() use ( $webfonts ) {
+					self::$used_webfonts = array_merge(
+						self::$used_webfonts,
+						$webfonts
+					);
+				}
+			);
+		}
+	}
+
+	/**
+	 * Get globally used webfonts.
+	 *
+	 * @return array
+	 */
+	private function get_globally_used_webfonts() {
+		$global_styles          = gutenberg_get_global_styles();
+		$globally_used_webfonts = array();
+
+		// Register used fonts from blocks.
+		foreach ( $global_styles['blocks'] as $id => $setting ) {
+			$font_family_slug = $this->get_font_family_from_setting( $setting );
+
+			if ( $font_family_slug ) {
+				$globally_used_webfonts[ $font_family_slug ] = 1;
```

Comment:
Curious why we use `1` instead of `true`?

Reply:
It's just shorter  🤷 

It doesn't make a difference because I'm using this as an object so I can make faster lookups instead of using an array.